cmd/shortener/main.go

```diff
@@ -3,42 +3,52 @@ package main
 import (
 	"log/slog"
 	"os"
+	"time"
 
 	"mishin-shortener/internal/app/config"
 	"mishin-shortener/internal/app/filestorage"
 	"mishin-shortener/internal/app/handlers"
 	"mishin-shortener/internal/app/mapstorage"
 	middleware "mishin-shortener/internal/app/midleware"
+	"mishin-shortener/internal/app/pgstorage"
+
+	chiMiddleware "github.com/go-chi/chi/v5/middleware"
 
 	"net/http"
 
 	"github.com/go-chi/chi/v5"
 )
 
+func initStorage(c config.MainConfig) handlers.AbstractStorage {
+	if c.DatabaseDSN != "" {
+		return pgstorage.Make(c)
+	}
+	if c.FileStoragePath != "" {
+		return filestorage.Make(c.FileStoragePath)
+	}
+
+	return mapstorage.Make()
+}
+
 func main() {
 	c := config.MakeConfig()
 	c.InitConfig()
 
-	var storage handlers.AbstractStorage
-
-	// Если файлик определен, то заведем файлохранилище. Иначе хватит просто мапы
-	if c.FileStoragePath != "" {
-		storage = filestorage.Make(c.FileStoragePath)
-	} else {
-		storage = mapstorage.Make()
-	}
-
+	storage := initStorage(c)
 	defer storage.Finish()
 
 	h := handlers.MakeShortanerHandler(c, storage)
 
 	r := chi.NewRouter()
 	r.Use(middleware.WithLogRequest)
 	r.Use(middleware.GzipMiddleware)
+	r.Use(chiMiddleware.Timeout(60 * time.Second))
 
-	r.Post("/", h.CreateURLHandler)
-	r.Post("/api/shorten", h.CreateURLByJSONHandler)
+	r.Post("/", h.CreateURL)
+	r.Post("/api/shorten", h.CreateURLByJSON)
+	r.Post("/api/shorten/batch", h.CreateURLByJSONBatch)
```

Comment:
Можно использовать функционал chi для объединения ручек. Это поможет сделать роутер немного более читаемым особенно случае, если маршрутов много. 
```suggestion
r.Route("/api", func(r chi.Router) {
	r.Post("/shorten", h.CreateURLByJSON)
	r.Post("/shorten/batch", h.CreateURLByJSONBatch)
}
```